GitCommands/Settings/AppSettings.cs

```diff
@@ -1215,7 +1215,29 @@ public static Font MonospaceFont
         public static Font Font
         {
             get => GetFont("font", SystemFonts.MessageBoxFont);
-            set => SetFont("font", value);
+            set
+            {
+                SetFont("font", value);
+                _fontByStyle.Clear();
+            }
+        }
+
+        private static Dictionary<FontStyle, Font> _fontByStyle = new Dictionary<FontStyle, Font>();
```

Comment:
I would have liked to have used an array indexed by FontStyle, except that FontStyle's values are bit masks. I could still use them, but we'd have a sparse array of size 8 where only 1, 2, 4, and 8 indices would be used. It would be faster than using a dictionary. Or we could also convert bit position to index, but that requires looping to count number of right bit shifts. Thoughts?